templates/federated_reporting/import.sh

```diff
@@ -100,6 +102,22 @@ else
   done
 fi
 
+if [ "$CFE_FR_HANDLE_DUPLICATES" = "yes" ]; then
+  log "Handle Duplicate Hostkeys"
+  hostkey_list=$(printf "'%s'," ${hostkeys[*]} | sed -e 's/,$//')
+  "$CFE_BIN_DIR"/psql -U $CFE_FR_DB_USER -d cfdb --set "ON_ERROR_STOP=1" \
+                      -c "SELECT handle_duplicate_hostkeys_in_import(ARRAY[$hostkey_list]);"  \
+    > duplicates.log || failed=1
+  if [ "$failed" = "0" ]; then
+    log "Handle Duplicate Hostkeys: DONE"
+  else
+    log "Handle Duplicate Hostkeys: FAILED"
+    log "last 10 lines of duplicates.log"
+    tail -n 10 duplicates.log
```

Comment:
How big is this file? Shall we delete it? Or should it be compressed?

Reply:
It shouldn't be very big. There is no logging, say, for each host involved so will only scale linearly compared to the number of feeder hubs. We should make it the most recent one available the same as we do for the sql logs.

Reply:
Interestingly, I don't see where we remove the `schema_setup.log` either. I think any other logs besides the big compressed log from the actual data import should be left around for the last run and removed just before the next run.

I'll check a running system to confirm how this is working.

Reply:
The file is very small and will be a constant size regardless of the number of records processed.

This along with the schema_setup.log and schema_attach.log will remain at `/opt/cfengine/federation/cfapache` and be re-written on next run. We might want to mention this location as a debugging aid as well as the tarred up import logs.

Reply:
Though I did need to add a `2>&1` so that stderr also goes to that logfile instead of the agent log.